codex/marketplace.md

```diff
@@ -0,0 +1,157 @@
+The goal is to create a storage marketplace that promotes durability.
+
+## Motivation
+The Codex network aims to create a peer-to-peer storage engine with strong data availability, 
+data persistence guarantees and node storage incentives.
+To reach this goal, a data availability and retrieval mechanism is needed.
+Support for light clients, like mobile devices should also be embraced.
+The protocol should remove complexity to allow for simple implementation and 
+simplify incentive mechanisms.
+
+## Semantics 
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+| Terminology  | Description |
+| --------------- | --------- |
+| Storage Nodes | A Codex node that provides storage services to the marketplace.|
+| Validator Nodes | A Codex node that collects, validates and submits proofs to reward or penalize other storage nodes or validator nodes. |
+| Regular Nodes | The main Codex client that interacts with other nodes to locate and retrieve data. Also considered to be an ephemeral node (light client) |
+| Slots | An agreement between storage nodes and regular nodes to store data |
+
+### Storage Request
+
+A regular node can request storage of data by opening a storage contract on the Codex blockchain. 
+The storage contract is opened in the marketplace via the blockchain, and 
+the requested data is split into blocks with erasure coding.
+Blocks of data are placed into slots to begin the request,
+see [slots](#slots) discussed below. 
+
+The requester MUST provide `duration` of the storage request along with the appropriate `reward`,
+payment for request, before the contract is created. 
+Once a contract is created, 
+all slots MUST be filled by storage nodes before the request is officially started.
+If the request does not attract enough storage nodes after a pre-defined network timeout,
+which is defined by a Codex node as `expire`,
+the request SHOULD be canceled.
+If canceled, `collateral` SHOULD be returned to any storage nodes and 
+`reward` returned to the requester.
+The requester MAY create a new request with different values to attract storage nodes.
+
+The requester SHOULD initiate the contract with the following values:
+
+```nim
+
+proc requestStorage (): {} =
+  # content identifier
+  byte32 cid = 1
+  # Tokens from the requester to reward storage nodes
+  UInt256 reward = 2
+  # Amount of tokens required for collateral by storage nodes
+  UInt256 collateral = 3
+  # Frequency that proofs are checked by validator nodes
+  byte proofProbability = 4
+  # Amount of desired time for stoageRequest
+  UInt256 duration = 5
+  # The requester can choose amount of spreading of data to storage nodes
+  uint dispersal = 6
+  # Minimum Number of storage hosts 
+  uint nodes = 7
```

Comment:
Are all these values required to be provided by the requester, or does a Codex node provide some default configurations?

Reply:
It is likely better to reference the `Request` data structure here, as a Codex client will need to submit a request for storage in the contract using this structure: https://github.com/codex-storage/codex-contracts-eth/blob/4d9320a58208f9d9609f6edc3cccdb29619e31ae/contracts/Requests.sol#L7-L12

These values are all required to be provided by the client.